### PR TITLE
fix: add negative timestamp support

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates to: #2589 

Add support for negative timestamps before 1970-01-01.

### CODE REVIEW

File: packages/odata-v2/src/de-serializers/converters.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function serializeFromTime(value: Time): string {
  */
 export function deserializeToMoment(edmDateTime: string): moment.Moment {
   const dateTimeOffsetComponents =
-    /^\/Date\((?<ticks>\d+)((?<sign>[+-])(?<offset>\d{4}))?\)\/$/.exec(
+    /^\/Date\((?<ticks>-{0,1}\d+)((?<sign>[+-])(?<offset>\d{4}))?\)\/$/.exec(
```
Comment (fwilhe) [inline packages/odata-v2/src/de-serializers/converters.ts:42]: We do already have a "sign" group. How does that relate to what we're changing here?

Comment (fwilhe) [reply packages/odata-v2/src/de-serializers/converters.ts:42]: Screenshot 2022-06-20 at 15 29 58

Not sure if this is an issue, but I find it at least confusing that we have a negative timestamp where the sign is `+`.

Comment (florian-richter) [reply packages/odata-v2/src/de-serializers/converters.ts:42]:
```suggestion
    /^\/Date\((?<signedticks>-{0,1}\d+)((?<offsetsign>[+-])(?<unsignedoffset>\d{4}))?\)\/$/.exec(
```
[pp] I don't mind the current implementation but maybe this improves clarity?

**Review by florian-richter — APPROVED** [review]
LGTM

File: packages/odata-v2/src/de-serializers/converters.ts (hunk 1 above)
Comment (fwilhe) [reply packages/odata-v2/src/de-serializers/converters.ts]: That makes it more clear to me, thanks

**Review by fwilhe — APPROVED** [review]

**Pull request merged**